site/include/form.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+
+$FormErrors = array();
+
+function TextBox($FieldName){
```

Comment:
If you add a second optional parameter here (I'd suggest using a [default function parameter for this](http://www.hashbangcode.com/blog/default-function-parameters-php)) then this should be able to replace the `EditTextBox` function. And as it is right now, this function considers the request, but `EditTextBox` doesn't, which means if you're editing a form and it doesn't validate, everything you entered will be lost.